### DISCUSSION TITLE
ImageMagick -format argument No decode delegate

### DISCUSSION THREAD

**Discussion by daniloak** [ImageMagick/ImageMagick#5695] [Help] [answered]
I am getting errors when trying to use the format argument in any command, I tried to espace the using backslashes like this `\"%m:%f %wx%h\"` but it doesn't work either

I am running on a MAC M1, not sure if it could impact

Example

    identify -units PixelsPerInch -format "%m:%f %wx%h"  238d99c5-8b58-46c7-ac94-b28b3898b8b8.jpeg

And I get 

    identify: unable to open image '%m:%f %wx%h': No such file or directory @ error/blob.c/OpenBlob/3570. 
    identify: no decode delegate for this image format `' @ error/constitute.c/ReadImage/741.

I checked the decodes installed and jpeg is there 

    Version: ImageMagick 7.1.0-51 Q16-HDRI aarch64 20499 https://imagemagick.org
    Copyright: (C) 1999 ImageMagick Studio LLC
    License: https://imagemagick.org/script/license.php
    Features: Cipher DPC HDRI Modules OpenMP(5.0) 
    Delegates (built-in): bzlib fontconfig freetype gslib heic jng jp2 jpeg lcms lqr ltdl lzma openexr png ps raw tiff webp xml zlib
    Compiler: gcc (4.2)

**Comment by dlemstra** [comment]
Are you sure this is a `jpeg` file and not a file pretending to be a `jpeg` file? I suspect that you have a `heic` file instead. What is the output of `file 238d99c5-8b58-46c7-ac94-b28b3898b8b8.jpeg`?

**Reply by daniloak** [reply]
```
file 238d99c5-8b58-46c7-ac94-b28b3898b8b8.jpeg
238d99c5-8b58-46c7-ac94-b28b3898b8b8.jpeg: JPEG image data, JFIF standard 1.01, resolution (DPI), density 150x150, segment length 16, baseline, precision 8, 5400x3601, components 3
```

**Reply by daniloak** [reply]
I don't think it's the image, it happens with any image after I put the format argument, without it works

**Comment by fmw42** [comment]
With Imagemgagick 7, use magick identify, not identify.

It works for me on IM 7.1.0.51

`magick identify -units PixelsPerInch -format "%m:%f %wx%h" lena.jpg`
JPEG:lena.jpg 256x256

Check you image? Is it valid?

**Reply by daniloak** [reply]
The image is valid. Using magick instead gives me the same error

**Reply by fmw42** [reply]
Did you use "magick" or "magick identify"? If the latter, then I think you might need to reinstall your Imagemagick.

**Reply by daniloak** [reply]
I tried to reinstall with brew, it's weird because it works well without -format

**Reply by fmw42** [reply]
Are you using curly quotes or straight quotes?

**Comment by daniloak** [comment, marked as answer]
I don't know what happened, I reinstalled everything, restarted and it is working
